src/DIRAC/TransformationSystem/DB/TransformationDB.py

```diff
@@ -160,33 +160,51 @@ def addTransformation(
         if not res["OK"]:
             return S_ERROR("Failed to parse the transformation body")
         body = res["Value"]
-        req = (
-            "INSERT INTO Transformations (TransformationName,Description,LongDescription, \
-                                        CreationDate,LastUpdate,AuthorDN,AuthorGroup,Type,Plugin,AgentType,\
-                                        FileMask,Status,TransformationGroup,GroupSize,\
-                                        InheritedFrom,Body,MaxNumberOfTasks,EventsPerTask)\
-                                VALUES ('%s','%s','%s',\
-                                        UTC_TIMESTAMP(),UTC_TIMESTAMP(),'%s','%s','%s','%s','%s',\
-                                        '%s','New','%s',%f,\
-                                        %d,%s,%d,%d);"
-            % (
-                transName,
-                description,
-                longDescription,
-                authorDN,
-                authorGroup,
-                transType,
-                plugin,
-                agentType,
-                fileMask,
-                transformationGroup,
-                groupSize,
-                inheritedFrom,
-                body,
-                maxTasks,
-                eventsPerTask,
-            )
-        )
+
+        escapable_columns = [
+            "TransformationName",
+            "Description",
+            "LongDescription",
+            "AuthorDN",
+            "AuthorGroup",
+            "Type",
+            "Plugin",
+            "AgentType",
+            "FileMask",
+            "Status",
+            "TransformationGroup",
+        ]
+
+        params = {
+            "TransformationName": transName,
+            "Description": description,
+            "LongDescription": longDescription,
+            "CreationDate": "UTC_TIMESTAMP()",
+            "LastUpdate": "UTC_TIMESTAMP()",
+            "AuthorDN": authorDN,
+            "AuthorGroup": authorGroup,
+            "Type": transType,
+            "Plugin": plugin,
+            "AgentType": agentType,
+            "FileMask": fileMask,
+            "Status": "New",
+            "TransformationGroup": transformationGroup,
+            "GroupSize": groupSize,
+            "InheritedFrom": inheritedFrom,
+            "Body": body,
+            "MaxNumberOfTasks": maxTasks,
+            "EventsPerTask": eventsPerTask,
+        }
+
+        for key in params:
+            if key in escapable_columns:
+                res = self._escapeString(params[key])
+                if not res["OK"]:
+                    return S_ERROR(f"Could not escape parameter {key}")
```

Comment:
maybe add the value that couldn't be escaped?

Reply:
I think it's safer not to broadcast the contents back to the user, in case the contents are sensitive. The parameter name should be enough to point debugging in the correct direction.

Reply:
Which of the escaped fields has potential sensitive content that the user oughtn't already know?
 
```suggestion
                    return S_ERROR(f"Could not escape value for parameter {key}")
```

Reply:
Perhaps none of the parameters are particularly sensitive, but I still lean towards not repeating the values back to the caller.

> user oughtn't already know

OK, so the user already knows the parameters. Why repeat them back? If it happens to be a long value, then it just makes the error messages more difficult to read.

Reply:
In case the value is _not_ what the user expects it to be.